securedrop/tests/test_source.py

```diff
@@ -558,6 +558,32 @@ def test_metadata_route(config, source_app):
             assert resp.json.get('server_os') == '16.04'
             assert resp.json.get('supported_languages') ==\
                 config.SUPPORTED_LOCALES
+            assert resp.json.get('v2_source_url') is None
+            assert resp.json.get('v3_source_url') is None
+
+
```

Comment:
We could add a test for `check_url_file`, e.g.:
```
def test_check_url_file(config):

    assert check_url_file("nosuchfile", "whatever") is None

    try:
        def write_url_file(path, content):
            url_file = open(path, "w")
            url_file.write("{}\n".format(content))

        url_path = "test_source_url"

        onion_test_url = "abcdabcdabcdabcd.onion"
        write_url_file(url_path, onion_test_url)
        assert check_url_file(url_path, r"^[a-z0-9]{16}\.onion$") == onion_test_url

        onion_test_url = "abcdefghabcdefghabcdefghabcdefghabcdefghabcdefghabcdefgh.onion"
        write_url_file(url_path, onion_test_url)
        assert check_url_file(url_path, r"^[a-z0-9]{56}\.onion$") == onion_test_url

        write_url_file(url_path, "NO.onion")
        assert check_url_file(url_path, r"^[a-z0-9]{56}\.onion$") is None
    finally:
        if os.path.exists(url_path):
            os.unlink(url_path)
```

Reply:
No test coverage of anything in source_app/utils.py, at a glance. Maybe a separate module for it?

Reply:
:open_mouth: Yeah, if so that should be corrected. And this test could be handled there.

Reply:
By "there" I meant in the later PR to correct the lack of utils tests.

Reply:
woops, I added it in this PR - but will stick in an issue to add more tests for other `utils` fns, as it currently only has yours above.